pennylane/pulse/parametrized_hamiltonian_pytree.py

```diff
@@ -92,12 +92,11 @@ def tree_flatten(self):
             tuple: tuple containing the matrices and the parametrized coefficients defining the class
         """
         matrices = (self.mat_fixed, self.mats_parametrized)
-        param_coeffs = self.coeffs_parametrized
-        reorder_fn = self.reorder_fn
-        return (matrices, param_coeffs, reorder_fn)
+        metadata = (self.coeffs_parametrized, self.reorder_fn)
+        return (matrices, metadata)
 
     @classmethod
-    def tree_unflatten(cls, param_coeffs: tuple, matrices: tuple, reorder_fn: callable):
+    def tree_unflatten(cls, metadata: tuple, matrices: tuple):
```

Comment:
this was a broken implementation... It's fixed now.

Reply:
also tagging @dwierichs fyi

Reply:
Thanks :)

Reply:
Might be good to add a test that would have detected this ?

Reply:
added